src/Components/Common/WizardContext/initialState.js

```diff
@@ -2,8 +2,11 @@ const initialWizardContext = {
   chosenSource: undefined,
   chosenNumOfInstances: 1,
   chosenInstanceType: undefined,
+  uploadedKey: false,
```

Comment:
is this flag needed? if the `chosenSshKeyId` is defined we can deduce it anyway, no?

Reply:
I don't think so, what happens if you choose a key, but then change your mind and upload it instead, we will have `chosenSshKeyId` defined, won't we?

Reply:
we can just reset the `chosenSshKeyId` back to undefined when the user uploads a key

Reply:
we can for sure, I just thought that would be much more corner cases to worry about, it is doable, but this felt bit more straight forward :shrug: